### PR TITLE
Feature/root node parameter

### PR DESCRIPTION
**Pull request by ziachap**

### CODE REVIEW

File: Gibe.Navigation.Umbraco/UmbracoNavigationProvider.cs [hunk 1]
```diff
@@ -10,90 +9,88 @@
...
+            _umbracoWrapper = umbracoWrapper;
+            Priority = priority;
+            _rootNode = rootNode;
+        }
+
+        public int Priority { get; }
+
+        public IEnumerable<T> GetNavigationElements()
+        {
+            var topLevel = _umbracoNodeService.GetNode(_rootNode);
```
Comment (ziachap) [inline Gibe.Navigation.Umbraco/UmbracoNavigationProvider.cs]: this is whats changed

File: Gibe.Navigation.Umbraco/DefaultNodeTypeFactory.cs [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+        {
+            _nodeTypes = nodeTypes.ToDictionary(n => n.GetType(), n => n);
+        }
+
+        public INodeType GetNodeType<T>() where T : INodeType
+        {
+            return _nodeTypes[typeof(T)];
+        }
+
+        public INodeType GetNodeType(Type T)
```
Comment (stevetemple) [inline Gibe.Navigation.Umbraco/DefaultNodeTypeFactory.cs]: lowercase T as a parameter

File: Gibe.Navigation.Umbraco/INodeTypeFactory.cs [hunk 3]
```diff
@@ -10,5 +10,7 @@ namespace Gibe.Navigation.Umbraco
 	public interface INodeTypeFactory
 	{
 		INodeType GetNodeType<T>() where T : INodeType;
+
+	    INodeType GetNodeType(Type T);
```
Comment (stevetemple) [inline Gibe.Navigation.Umbraco/INodeTypeFactory.cs]: lowercase T

File: Gibe.Navigation.Umbraco/UmbracoNavigationProvider.cs [hunk 4]
```diff
@@ -12,28 +12,30 @@ namespace Gibe.Navigation.Umbraco
...
+        private readonly Type _rootNode;
         private readonly IUmbracoNodeService _umbracoNodeService;
         private readonly IUmbracoWrapper _umbracoWrapper;
 
         public UmbracoNavigationProvider(
             IModelConverter modelConverter,
             IUmbracoNodeService umbracoNodeService,
             IUmbracoWrapper umbracoWrapper,
-            int priority, INodeType rootNode)
+            int priority, Type rootNode, INodeTypeFactory nodeTypeFactory)
```
Comment (stevetemple) [inline Gibe.Navigation.Umbraco/UmbracoNavigationProvider.cs]: swap the order of these new ones

Comment (stevetemple) [inline Gibe.Navigation.Umbraco/UmbracoNavigationProvider.cs]: Also change rootNode to rootNodeType

**Pull request merged**